Automattic/jetpack
# Fix Super Cache / Jetpack Compatibility

Fixes https://github.com/Automattic/jpop-issues/issues/7868

A fatal error can occur when running Jetpack 11.5 and Super Cache side-by-side, if Super Cache has its "Jetpack Mobile Theme" plugin enabled. This was caused by a change to how the device-detection stuff in Jetpack 11.5 was autoloaded.

Things I'm uncertain of:
- Does adding device-detection as a dependency of Super Cache require anything in the repository setup to ensure the vendor stuff will get added to the next deployed version?

#### Changes proposed in this Pull Request:
* Add Jetpack's device detection as a dependency of Super Cache, so that it can rely on its own copy
* Ensure that the User_Agent class is automatically loaded if not run in an auto-load environment in device-detection.

#### Other information:

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Install Jetpack 11.5 and Super Cache side-by-side
* Enable Jetpack mobile Theme under the "Plugins" tab of Super Cache
* Load your home page in a private browser window
* Ensure that it no longer crashes :)
